gnomad_qc/v3/subset/subset.py

```diff
@@ -0,0 +1,292 @@
+    :param partition_size: Amount of data per partition. Hail recommends 128MB, defaults to 128000000
+    :return: number of partitions for resizing MT  
+    """
+    rows, columns = mt.count()
+    mt_disk_est = rows * columns * entry_size
+    n_partitions = (
+        int(mt_disk_est / partition_size) if (mt_disk_est / partition_size) > 20 else 20 # make max
+    )
+    return n_partitions # bring to gnomad_methods -> parameterize
+
+
+def main(args):
+    hl.init(log="/subset.log", default_reference="GRCh38")
+    pop = args.pop
+    subset = args.subset
+    hard_filter = args.hard_filter_samples
+    release = args.release
+    sparse = args.sparse
+    header_dict = HEADER_DICT
+
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, remove_hard_filtered_samples=hard_filter, release_only=release
+    )
+    info_ht = get_info().ht()
+    info_ht = format_info_for_vcf(info_ht)
+
+    if "SB" in info_ht.info and not isinstance(
+        info_ht.info.SB, hl.expr.ArrayNumericExpression
+    ):
+        info_ht = info_ht.annotate(
+            info=info_ht.info.annotate(SB=hl.flatten(info_ht.info.SB))
+        )
+
+    meta = metadata.ht()
+
+    if pop and pop in GENOME_POPS:
+        logger.info(f"Subsetting samples to {pop} population")
+        mt = mt.filter_cols(meta[mt.col_key].pop == pop)
+
+    if subset:
+        gt_expr = "LGT" if sparse else "GT" #assumes matrix table is not split - infer 
+        mt = subset_samples_and_variants(mt, subset, sparse=sparse, gt_expr=gt_expr)
+
+    if args.add_meta:
+        meta = meta.semi_join(mt.cols())
+        meta.export(f"{args.output_path}/metadata.tsv.bgz")
+
+    if sparse:
+        mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
+        mt = hl.experimental.densify(mt)
+    else:
+        mt = hl.split_multi_hts(mt)
+
+    mt = mt.filter_rows(hl.len(mt.alleles) > 1)
+    mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+
+    mt = mt.drop(mt.gvcf_info)
+    mt = mt.annotate_rows(info=info_ht[mt.row_key].info)
+
+    if args.subset_call_stats:
+        mt = mt.annotate_rows(subset_callstats=hl.agg.call_stats(mt.GT,mt.alleles))     
+        mt = mt.annotate_rows(info=mt.info.annotate(subset_AC=mt.subset_callstats.AC[1],
+                                                    subset_AN=mt.subset_callstats.AN,
+                                                    subset_AF=hl.float32(mt.subset_callstats.AF[1])))
+        mt = mt.drop('subset_callstats')
+        header_dict['info'].update({"subset_AC": {"Description": "Alternate Allele count in subset post-filtering"} , 
+                                    "subset_AN": {"Description": "Ref and Alternate allele number in subset post-filtering"},
+                                    "subset_AF": {"Description": "Alternate Allele frequency in subset post-filtering"}})
+    
+    if args.add_gnomad_freqs:
+        from gnomad_qc.v3.resources.annotations import freq
```

Comment:
To not crowd the namespace with unnecessary variables/methods but I'm not sure it's really necessary since this is not yet generalized for wider use. Im going to move it up.

Reply:
Thank you!